modular_zubbers/code/modules/rimiprefs/code/helpers.dm

```diff
@@ -0,0 +1,63 @@
+/// An inexact color darken proc. Do not use if you need precise adjustments.
+/proc/darken_color(color_to_darken)
+	var/new_color = ""
+	var/char
+	var/ascii
+	for(var/index = 1, index <= length(color_to_darken), index += length(char))
+		char = color_to_darken[index]
+		ascii = text2ascii(char)
+		switch(ascii)
+			if(48)
+				new_color += "0"
+			if(49 to 57)
+				new_color += ascii2text(ascii-1) //numbers 1 to 9
+			if(97)
+				new_color += "9"
+			if(98 to 102)
+				new_color += ascii2text(ascii-1) //letters b to f lowercase
+			if(65)
+				new_color += "9"
+			if(66 to 70)
+				new_color += ascii2text(ascii+31) //letters B to F - translates to lowercase
+			else
+				break
+
+	return new_color
```

Comment:
This is utterly cursed, do the colors we use this on pass through rgb2num? 
If so is it possible to work with the color after it's made into the color matrix which would make this easier and cheaper?

(Sorry for not checking this myself)

Reply:
New darken method done, and looks pretty close to the old system, so there we go.

![image](https://github.com/user-attachments/assets/f77581fb-3a93-41a4-a421-bffcdf0896b8)